syntax = "proto3";

package app.ping.v1;

message PingRequest {
    string target = 10;
    int32 delay = 20;
}
message PingReply {
    bool ok = 1;
    string msg = 2;
    int32 delay = 3;
}

message HelloRequest {
    string username = 1;
}
message HelloReply {
    bool ok = 1;
    int32 value = 2;
}

service PingService {
    rpc Ping (PingRequest) returns (PingReply);
}

service HelloService {
    rpc Hello (HelloRequest) returns (HelloReply);
}